lib/private/Log/ErrorHandler.php

```diff
@@ -27,84 +30,93 @@
+								bool $debug) {
+		$this->logger = $logger;
 
 		if ($debug) {
-			set_error_handler([$handler, 'onAll'], E_ALL);
-			if (\OC::$CLI) {
-				set_exception_handler(['OC_Template', 'printExceptionErrorPage']);
+			set_error_handler(Closure::fromCallable([$this, 'onAll']), E_ALL);
+			if ($isCli) {
+				set_exception_handler(Closure::fromCallable(['OC_Template', 'printExceptionErrorPage']));
 			}
 		} else {
-			set_error_handler([$handler, 'onError']);
+			set_error_handler(Closure::fromCallable([$this, 'onError']));
 		}
-		register_shutdown_function([$handler, 'onShutdown']);
-		set_exception_handler([$handler, 'onException']);
+		register_shutdown_function(Closure::fromCallable([$this, 'onShutdown']));
+		set_exception_handler(Closure::fromCallable([$this, 'onException']));
 	}
 
-	public static function setLogger(ILogger $logger) {
-		self::$logger = $logger;
+	/**
+	 * remove password in URLs
+	 * @param string $msg
+	 * @return string
+	 */
+	private static function removePassword($msg) {
+		return preg_replace('#//(.*):(.*)@#', '//xxx:xxx@', $msg);
 	}
 
-	//Fatal errors handler
-	public static function onShutdown() {
+	/**
+	 * Fatal errors handler
+	 */
+	private function onShutdown(): void {
 		$error = error_get_last();
-		if ($error && self::$logger) {
-			//ob_end_clean();
+		if ($error) {
 			$msg = $error['message'] . ' at ' . $error['file'] . '#' . $error['line'];
-			self::$logger->critical(self::removePassword($msg), ['app' => 'PHP']);
+			$this->logger->critical(self::removePassword($msg), ['app' => 'PHP']);
 		}
 	}
 
 	/**
-	 * 	Uncaught exception handler
-	 *
-	 * @param \Exception $exception
+	 * Uncaught exception handler
 	 */
-	public static function onException($exception) {
+	private function onException(Throwable $exception): void {
 		$class = get_class($exception);
 		$msg = $exception->getMessage();
 		$msg = "$class: $msg at " . $exception->getFile() . '#' . $exception->getLine();
-		self::$logger->critical(self::removePassword($msg), ['app' => 'PHP']);
+		$this->logger->critical(self::removePassword($msg), ['app' => 'PHP']);
 	}
 
-	//Recoverable errors handler
-	public static function onError($number, $message, $file, $line) {
+	/**
+	 * Recoverable errors handler
+	 */
+	private function onError($number, $message, $file, $line): void {
 		if (!(error_reporting() & $number)) {
 			return;
 		}
 		$msg = $message . ' at ' . $file . '#' . $line;
-		$e = new \Error(self::removePassword($msg));
-		self::$logger->logException($e, ['app' => 'PHP', 'level' => self::errnoToLogLevel($number)]);
+		$e = new Error(self::removePassword($msg));
+		$this->logger->log(self::errnoToLogLevel($number), $e->getMessage(), ['app' => 'PHP']);
 	}
 
-	//Recoverable handler which catch all errors, warnings and notices
-	public static function onAll($number, $message, $file, $line) {
+	/**
+	 * Recoverable handler which catch all errors, warnings and notices
+	 */
+	private function onAll($number, $message, $file, $line): void {
```

Comment:
```suggestion
	 * Recoverable handler which catch all errors, warnings and notices
	 	 * Recoverable handler which catch all errors, warnings and notices
	 * @param int $number
	 * @param string $message
	 * @param string $file
	 * @param int $line
	 * @paran array $context
	 */
	private function onAll($number, $message, $file, $line, $context): void {
```

Hopefully tell psalm to be more quiet https://www.php.net/manual/en/function.set-error-handler.php

Reply:
I went will full typing :popcorn: